lib/diagnostic.js

```diff
@@ -8,22 +8,41 @@
 // and returns the minimum from that set
 // assume that all the argument are numbers
 const min = function min() {
+  let mini = arguments[0];
 
+  for (let i = 0; i < arguments.lenth; i++) {
```

Comment:
This typo causes the loop to never run.  As it is, it evaluates ```i < arguments.lenth;``` as ```0 < undefined```, which evaluates to false.